src/holon/models/rule_actions/rule_action_add_set.py

```diff
@@ -176,6 +180,101 @@ def add_or_set_items(
 
                 objects_added += 1
 
+    def add_or_set_items(
+        self,
+        scenario_aggregate: ScenarioAggregate,
+        filtered_repository: RepositoryBaseClass,
+        value: str,
+        reset_models_before_add: bool,
+    ) -> ScenarioAggregate:
+        """Add an asset to the first n items in the the filtered objects"""
+
+        if filtered_repository.len() <= 0:
+            return scenario_aggregate
+
+        if reset_models_before_add:
+            # parse value
+            n = int(float(value))
+            if n < 0:
+                raise ValueError(f"Value to add cannot be smaller than 0. Given value: {n}")
+
+        # get parent type and foreign key field name
+        base_parent_type = filtered_repository.base_model_type  # TODO make sure repository has this
+        try:
+            parent_fk_field_name = next(
+                parent_fk_fieldname
+                for parent_type, parent_fk_fieldname in RuleActionUtils.get_parent_classes_and_field_names(
+                    self.model_to_add.__class__
+                )
+                if base_parent_type == parent_type
+            )
+        except:
+            raise ValueError(
+                f"Type {base_parent_type} in the filter does not match found parent type {RuleActionUtils.get_parent_classes_and_field_names(self.model_to_add.__class__)} for model type {self.model_to_add.__class__.__name__}"
+            )
+
+        objects_added = 0
+
+        # NOT NECESSARY ANYMORE? PLS REVIEW - TAVM
```

Comment:
That should be right. Since id's are stable now we shouldn't have to look for the new id.
When the tests are fixed we'll be absolutely sure